File: fe/src/app/services/passed-poll.service.ts
import {Injectable} from '@angular/core';
import {HttpClient, HttpParams} from '@angular/common/http';
import {PassedPoll} from '../modules/poll/models/passed-poll';
import {Observable} from 'rxjs';

@Injectable({providedIn: 'root'})
export class PassedPollService {

  private passedPollsUrl = '/api/passedPolls';  // URL to web api

  constructor(private http: HttpClient) {
  }

  passPoll(passedPoll: PassedPoll): Observable<any> {
    return this.http.post(this.passedPollsUrl, passedPoll);
  }

  getPassedPollsByPollId(pollId: string): Observable<any> {
    const params = new HttpParams({
      fromObject: {
        pollId
      }
    });
    return this.http.get(this.passedPollsUrl, {params});
  }
}

File: be/src/main/java/com/yanherasimau/be/service/PassedPollService.java
package com.yanherasimau.be.service;

import com.yanherasimau.be.entity.PassedPoll;

public interface PassedPollService {
    PassedPoll save(PassedPoll passedPoll);

    Iterable<PassedPoll> getPassedPollsByPollId(long pollId);
}
